examples/voting/src/lib.rs

```diff
@@ -0,0 +1,415 @@
+//! A voting smart contract example.
+//!
+//! # Description
+//! A contract that allows for conducting an election with several voting
+//! options. An end_time is set when the election is initialized. Only accounts
+//! can vote for one of the voting options. Each account can change its
+//! selected voting option as often as it desires until the end_time is reached.
+//! No voting will be possible after the end_time.
+//!
+//! # Operations
+//! The contract allows for
+//!  - `initializing` the election;
+//!  - `vote` for one of the voting options;
+//!  - `getNumberOfVotes` for each voting option;
+//!  - `view` general information about the election.
+//!
+//! Note: Vec<VotingOption> (among other variables) is an input parameter to the
+//! `init` function as well as a return_value (among other variables) of the
+//! `view` function. Since there is a limit to the parameter/return_value size,
+//! the size of the Vec<VotingOption> is limited.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+use concordium_std::*;
+
+type VotingOption = String;
+type VoteIndex = u32;
+type VoteCount = u32;
+
+/// The parameter type for the contract function `vote`.
+/// Takes a vote_index that the account wants to vote for.
+#[derive(Deserial, SchemaType)]
+struct VoteParameter {
```

Comment:
The front-end had that problem that with the given schema and passing in 0 as voting_index, you couldn't vote for the first option. There seems to be a bug with the schema decoding to handle the the edge case where the input parameter is 0. I suggest we should open a bug report for this and investigate it further.

For now, I use this struct because a struct is correctly decoded given the schema (even if it has a 0 `VoteIndex`)

Reply:
Ok, that sounds very odd. Yes, we'll definitely have to look into that.

Reply:
Where is this bug with the decoding visible? In cargo-concordium?

Reply:
If you use `VoteIndex` directly instead of using the struct `VoteParameter` in the function `vote`.
#[receive(
    contract = "voting",
    name = "vote",
    mutable,
    enable_logger,
    **parameter = "VoteIndex",**
    error = "VotingError"
)]

Then you get the following output when voting for the first option in the browser wallet:

Not correctly decoded (browser wallet does not show voteIndex 0):
![VoteIndex0IsNotDecoded](https://user-images.githubusercontent.com/13039169/202294765-05b63bbb-4761-4a9d-bfb2-61d2ab812a8e.png)

That is how the other voteIndexes are displayed:
![VoteIndex1IsCorrectlyDecoded](https://user-images.githubusercontent.com/13039169/202294668-d2531682-1853-4031-a78e-addda6b59a06.png)

This is the line at the front-end:
Using `{vote}` at the front-end at line L119 with `VoteParameter` as input parameter schema works:
Using `vote` at the front-end at line L119 with `VoteIndex` as input parameter schema works not if the user selects a 0 voteIndex:


https://github.com/Concordium/concordium-browser-wallet/blob/8a4b7c85c1b73aae5eb7a979b2ce1f48caece584/examples/voting/src/Wallet.js#L119

Reply:
Issue can be tracked here:
https://github.com/Concordium/concordium-browser-wallet/issues/189